src/ucp/wireup/select.c

```diff
@@ -2214,7 +2280,11 @@ ucp_wireup_select_lanes(ucp_ep_h ep, unsigned ep_init_flags,
     }
 
 out:
-    ucp_wireup_construct_lanes(&select_params, &select_ctx, addr_indices, key);
+    status = ucp_wireup_construct_lanes(&select_params, &select_ctx,
+                                        addr_indices, key);
+    if (status != UCS_OK) {
+        return status;
+    }
```

Comment:
is it unrelated bug fix?

Reply:
one of the gtests replaces get_device_address by a stub that returns error and expects wireup to fail if this stub is called at least once. so this change makes sure we fail whenever get_device_address fails from the context of the new locality code